libra-node/src/main_node.rs

```diff
@@ -148,6 +135,33 @@ pub fn setup_network(
             .take_private()
             .expect("Failed to take Network identity private key, key absent or already read");
         let identity_public = config.network_keypairs.identity_keys.public().clone();
+        let trusted_peers = if role == RoleType::Validator {
+            // for validators, initial trusted peers only include self and will be populated from consensus
+            let mut self_peers = HashMap::new();
+            self_peers.insert(
+                config.peer_id,
+                NetworkPublicKeys {
+                    signing_public_key: signing_public.clone(),
+                    identity_public_key: identity_public.clone(),
+                },
+            );
+            self_peers
+        } else {
+            config
+                .network_peers
+                .peers
+                .iter()
+                .map(|(peer_id, keys)| {
+                    (
+                        *peer_id,
+                        NetworkPublicKeys {
+                            signing_public_key: keys.network_signing_pubkey.clone(),
+                            identity_public_key: keys.network_identity_pubkey.clone(),
+                        },
+                    )
+                })
+                .collect()
+        };
```

Comment:
Seems weird that we cannot just leverage what the config is giving us directly. Also why do we want ourselves in the trusted set? We would get there after genesis executes anyway.